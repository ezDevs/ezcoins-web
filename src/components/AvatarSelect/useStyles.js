import { makeStyles } from '@material-ui/core';

export default makeStyles(() => ({
  avatar: {
    margin: 10,
    height: 80,
    width: 80,
    transition: 'all .2s ease-in-out',
    '&:hover, &.selected': {
      transform: 'scale(1.3)',
      cursor: 'pointer',
    },
  },
  container: {
    marginTop: 15,
    marginBottom: 50,
  },
}));
