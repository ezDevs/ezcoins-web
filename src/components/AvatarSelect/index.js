import React, { memo, useCallback } from 'react';
import { Avatar, Grid } from '@material-ui/core';
import * as PropTypes from 'prop-types';
import { map } from 'ramda';

import useStyles from './useStyles';

const AvatarSelect = memo(({ data, name, onChange, value }) => {
  const classes = useStyles();

  const onClick = useCallback(({ target: { alt } }) => onChange(name, alt), [name, onChange]);

  return (
    <Grid container justify="center" spacing={2} className={classes.container}>
      {map(
        ({ id, avatar }) => (
          <Grid key={id}>
            <Avatar
              alt={id}
              onClick={onClick}
              name={name}
              value={value}
              className={`${classes.avatar} ${id === value ? 'selected' : ''}`}
              src={avatar}
            />
          </Grid>
        ),
        data,
      )}
    </Grid>
  );
});

AvatarSelect.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      email: PropTypes.string,
    }),
  ),
  name: PropTypes.string,
  onChange: PropTypes.func,
  value: PropTypes.string,
};

AvatarSelect.defaultProps = {
  data: [],
  name: '',
  onChange: () => {},
  value: '',
};

export default AvatarSelect;
