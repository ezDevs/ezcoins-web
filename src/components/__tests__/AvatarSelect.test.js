import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import AvatarSelect from '../AvatarSelect';

describe('testing avatar select component', () => {
  it('should renders without props as expected', () => {
    const { container } = render(<AvatarSelect />);
    expect(container).toMatchSnapshot();
  });

  const id = '1';
  const name = 'testName';
  const onClick = jest.fn();
  let avatarSelect;

  beforeEach(() => {
    avatarSelect = render(
      <AvatarSelect
        name={name}
        data={[
          { id: '1', avatar: 'https://upload.wikimedia.org/wikipedia/pt/c/ce/Aang.png' },
          { id: '2', avatar: 'https://upload.wikimedia.org/wikipedia/pt/c/ce/Aang.png' },
        ]}
        onChange={onClick}
      />,
    );
  });

  it('should renders as expected', () => {
    const { container } = avatarSelect;
    expect(container).toMatchSnapshot();
  });

  it('should calls function onClick on select some avatar', () => {
    const { getByAltText } = avatarSelect;
    fireEvent.click(getByAltText(id));
    expect(onClick.mock.calls.length).toBe(1);
    expect(onClick.mock.calls[0]).toEqual([name, id]);
  });
});
