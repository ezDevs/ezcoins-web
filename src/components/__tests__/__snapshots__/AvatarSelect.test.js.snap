// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing avatar select component should renders as expected 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-container-114 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-113 "
        name="testName"
        value=""
      >
        <img
          alt="1"
          class="MuiAvatar-img"
          src="https://upload.wikimedia.org/wikipedia/pt/c/ce/Aang.png"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-113 "
        name="testName"
        value=""
      >
        <img
          alt="2"
          class="MuiAvatar-img"
          src="https://upload.wikimedia.org/wikipedia/pt/c/ce/Aang.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`testing avatar select component should renders without props as expected 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-container-2 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  />
</div>
`;
